File: src/my_py_pkg/my_py_pkg/add_two_ints_client.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

class MyNode(Node):
    def __init__(self):
        super().__init__('add_two_ints_client')
        self.client = self.create_client(AddTwoInts, 'add_two_ints')
    
    def add_two_ints(self, a: int, b: int):
        while not self.client.wait_for_service(timeout_sec=1.0):
            self.get_logger().info('Service "add_two_ints" not available, waiting again...')

        request = AddTwoInts.Request(a=a, b=b)
        future = self.client.call_async(request)
        future.add_done_callback(self.callback_add_two_ints)

    def callback_add_two_ints(self, future):
        try:
            response = future.result()
            self.get_logger().info(f'Result of add_two_ints: {response.sum}!')
        except Exception as e:
            self.get_logger().error(f'Service call failed: {e}')


def main(args=None):
    rclpy.init(args=args)
    node = MyNode()

    node.add_two_ints(3, 5)
    node.add_two_ints(31, 3)
    node.add_two_ints(1, 8)
    rclpy.spin(node)

    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: src/my_cpp_pkg/src/add_two_ints_client.cpp
#include <chrono>
#include <functional>
#include <memory>

#include "example_interfaces/srv/add_two_ints.hpp"
#include "rclcpp/rclcpp.hpp"

using namespace std::chrono_literals;

class AddTwoIntsClientNode : public rclcpp::Node
{
public:
  AddTwoIntsClientNode()
  : Node("add_two_ints_client")
  {
    client_ = this->create_client<example_interfaces::srv::AddTwoInts>("add_two_ints");
  }

  void add_two_ints(int64_t a, int64_t b)
  {
    while (!client_->wait_for_service(1s)) {
      if (!rclcpp::ok()) {
        RCLCPP_ERROR(this->get_logger(), "Interrupted while waiting for the service.");
        return;
      }

      RCLCPP_INFO(
        this->get_logger(),
        "Service \"add_two_ints\" not available, waiting again...");
    }

    auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();
    request->a = a;
    request->b = b;

    client_->async_send_request(
      request,
      std::bind(&AddTwoIntsClientNode::callback_add_two_ints, this, std::placeholders::_1));
  }

private:
  using AddTwoInts = example_interfaces::srv::AddTwoInts;
  using SharedFuture = rclcpp::Client<AddTwoInts>::SharedFuture;

  rclcpp::Client<AddTwoInts>::SharedPtr client_;

  void callback_add_two_ints(SharedFuture future)
  {
    try {
      auto response = future.get();
      RCLCPP_INFO(this->get_logger(), "Result of add_two_ints: %ld!", response->sum);
    } catch (const std::exception & error) {
      RCLCPP_ERROR(this->get_logger(), "Service call failed: %s", error.what());
    }
  }
};

int main(int argc, char ** argv)
{
  rclcpp::init(argc, argv);

  auto node = std::make_shared<AddTwoIntsClientNode>();
  node->add_two_ints(3, 5);
  node->add_two_ints(31, 3);
  node->add_two_ints(1, 8);

  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}
